unittests.py

```diff
@@ -183,6 +186,32 @@ def testOutputFileFromSourcefile(self):
         self._testFull(['/c', '/P', 'main.cpp'],
                        AnalysisResult.Ok, 'main.cpp', 'main.i')
 
+    def testPreprocessIgnoresOtherArguments(self):
+        # All those inputs must ignore the /Fo, /Fa and /Fm argument according
+        # to the documentation of /E, /P and /EP
+
+        # to file (/P)
+        self._testPreprocessingOutfile(['/c', '/P', 'main.cpp'], 'main.i')
+        self._testPreprocessingOutfile(['/c', '/P', '/FoSome.obj', 'main.cpp'], 'main.i')
+        self._testPreprocessingOutfile(['/c', '/P', '/FaListing.asm', 'main.cpp'], 'main.i')
+        self._testPreprocessingOutfile(['/c', '/P', '/FmMapfile.map', 'main.cpp'], 'main.i')
+
+        # to stdout (/E)
+        self._testPreprocessingOutfile(['/c', '/E', 'main.cpp'], None)
+        self._testPreprocessingOutfile(['/c', '/E', '/FoSome.obj', 'main.cpp'], None)
+        self._testPreprocessingOutfile(['/c', '/E', '/FaListing.asm', 'main.cpp'], None)
+        self._testPreprocessingOutfile(['/c', '/E', '/FmMapfile.map', 'main.cpp'], None)
+        # to stdout (/EP)
+        self._testPreprocessingOutfile(['/c', '/EP', 'main.cpp'], None)
+        self._testPreprocessingOutfile(['/c', '/EP', '/FoSome.obj', 'main.cpp'], None)
+        self._testPreprocessingOutfile(['/c', '/EP', '/FaListing.asm', 'main.cpp'], None)
+        self._testPreprocessingOutfile(['/c', '/EP', '/FmMapfile.map', 'main.cpp'], None)
+        # to stdout (/P /EP)
+        self._testPreprocessingOutfile(['/c', '/P', '/EP', 'main.cpp'], None)
+        self._testPreprocessingOutfile(['/c', '/P', '/EP', '/FoSome.obj', 'main.cpp'], None)
+        self._testPreprocessingOutfile(['/c', '/P', '/EP', '/FaListing.asm', 'main.cpp'], None)
+        self._testPreprocessingOutfile(['/c', '/P', '/EP', '/FmMapfile.map', 'main.cpp'], None)
```

Comment:
Nice one, thanks! I have the same behavior using MSVS2015. /P is with #line annotations and /P /EP is without – both go to file.